lib/HLSL/DxilModule.cpp

```diff
@@ -83,7 +69,22 @@ DxilModule::DxilModule(Module *pModule)
 , m_TessellatorPartitioning(DXIL::TessellatorPartitioning::Undefined)
 , m_TessellatorOutputPrimitive(DXIL::TessellatorOutputPrimitive::Undefined)
 , m_MaxTessellationFactor(0.f)
-, m_RootSignature(nullptr) {
+, m_Ctx(pModule->getContext())
+, m_pModule(pModule)
+, m_pEntryFunc(nullptr)
+, m_pPatchConstantFunc(nullptr)
+, m_EntryName("")
+, m_pMDHelper(llvm::make_unique<DxilMDHelper>(pModule, llvm::make_unique<DxilExtraPropertyHelper>(pModule)))
+, m_pDebugInfoFinder(nullptr)
+, m_pSM(nullptr)
+, m_DxilMajor(DXIL::kDxilMajor)
+, m_DxilMinor(DXIL::kDxilMinor)
+, m_ValMajor(1)
+, m_ValMinor(0)
+, m_pOP(llvm::make_unique<OP>(pModule->getContext(), pModule))
+, m_pTypeSystem(llvm::make_unique<DxilTypeSystem>(pModule))
+, m_pViewIdState(llvm::make_unique<DxilViewIdState>(this)) {
+
```

Comment:
Feel free to make sure, but this should just be a resorting. So no need to look for deeper meaning if you want to trust me.

Reply:
LGTM!